packages/playout-gateway/src/tsrHandlerJobs/initDeviceJob.ts

```diff
@@ -0,0 +1,201 @@
+import { Logger } from 'winston'
+import { Job } from './job'
+import {
+	AbortError,
+	CommandReport,
+	DeviceOptionsAny,
+	DeviceOptionsAtem,
+	DeviceStatus,
+	DeviceType,
+	MediaObject,
+	SlowFulfilledCommandInfo,
+	SlowSentCommandInfo,
+	StatusCode,
+} from 'timeline-state-resolver'
+import { TSRHandler } from '../tsrHandler'
+import { StatusObject } from '@sofie-automation/shared-lib/dist/peripheralDevice/peripheralDeviceAPI'
+import _ = require('underscore')
+import { CreateDeviceJobsResult } from './createDeviceJob'
+import { disableAtemUpload } from '../config'
+import { FinishedTrace, sendTrace } from '../influxdb'
+import debug = require('debug')
+
+export class InitDeviceJob extends Job<CreateDeviceJobsResult, CreateDeviceJobsResult, undefined> {
+	protected artifacts: undefined
+
+	constructor(
+		private deviceId: string,
+		private deviceOptions: DeviceOptionsAny,
+		private tsrHandler: TSRHandler,
+		private logger: Logger
+	) {
+		super()
+	}
+	async run(previousResult: CreateDeviceJobsResult, abortSignal?: AbortSignal): Promise<CreateDeviceJobsResult> {
+		if (abortSignal?.aborted) {
+			throw new AbortError()
+		}
+
+		const { deviceContainer, coreTsrHandler } = previousResult
+		// Set up device status
+		const deviceType = deviceContainer.deviceType
+
+		const onDeviceStatusChanged = (connectedOrStatus: Partial<DeviceStatus>) => {
+			let deviceStatus: Partial<StatusObject>
+			if (_.isBoolean(connectedOrStatus)) {
+				// for backwards compability, to be removed later
+				if (connectedOrStatus) {
+					deviceStatus = {
+						statusCode: StatusCode.GOOD,
+					}
+				} else {
+					deviceStatus = {
+						statusCode: StatusCode.BAD,
+						messages: ['Disconnected'],
+					}
+				}
+			} else {
+				deviceStatus = connectedOrStatus
+			}
+			coreTsrHandler.statusChanged(deviceStatus)
+
+			// When the status has changed, the deviceName might have changed:
+			deviceContainer.reloadProps().catch((err) => {
+				this.logger.error(`Error in reloadProps: ${err}`)
+			})
+			// hack to make sure atem has media after restart
```

Comment:
Since it was necessary to leave a comment explaining a portion of the method it indicates that said portion should instead be extracted into a method. Then you don't need the comment because the name of the extracted method will serve as the comment.